### PR TITLE
Fix insecure links

### PR DESCRIPTION
**Pull request by rex4539**

### CODE REVIEW

File: table/format.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ class Footer {
 };
 
 // kTableMagicNumber was picked by running
-//    echo http://code.google.com/p/leveldb/ | sha1sum
+//    echo https://github.com/google/leveldb | sha1sum
```
Comment (daira) [inline table/format.h:74]: Revert this change; the hash depends on that specific URL.

**Review by daira — CHANGES_REQUESTED** [review]
I checked all of these URLs. The change to the SHA-1 hash input should be reverted.

**Comment by daira** [comment]
Looks good to me.

**Review by daira — APPROVED** [review]

File: doc/benchmark.html [hunk 2]
```diff
@@ -83,7 +83,7 @@ <h1>LevelDB Benchmarks</h1>
 <p>Google, July 2011</p>
 <hr>
 
-<p>In order to test LevelDB's performance, we benchmark it against other well-established database implementations. We compare LevelDB (revision 39) against <a href="http://www.sqlite.org/">SQLite3</a> (version 3.7.6.3) and <a href="http://fallabs.com/kyotocabinet/spex.html">Kyoto Cabinet's</a> (version 1.2.67) TreeDB (a B+Tree based key-value store). We would like to acknowledge Scott Hess and Mikio Hirabayashi for their suggestions and contributions to the SQLite3 and Kyoto Cabinet benchmarks, respectively.</p>
+<p>In order to test LevelDB's performance, we benchmark it against other well-established database implementations. We compare LevelDB (revision 39) against <a href="https://www.sqlite.org/">SQLite3</a> (version 3.7.6.3) and <a href="https://dbmx.net/kyotocabinet/spex.html">Kyoto Cabinet's</a> (version 1.2.67) TreeDB (a B+Tree based key-value store). We would like to acknowledge Scott Hess and Mikio Hirabayashi for their suggestions and contributions to the SQLite3 and Kyoto Cabinet benchmarks, respectively.</p>
```
Comment (pwnall) [inline doc/benchmark.html:86]: Verified that `http://fallabs.com/kyotocabinet/spex.html` redirects to `https://dbmx.net/kyotocabinet/spex.html`.

File: doc/impl.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## Files
 
 The implementation of leveldb is similar in spirit to the representation of a
-single [Bigtable tablet (section 5.3)](http://research.google.com/archive/bigtable.html).
+single [Bigtable tablet (section 5.3)](https://research.google/pubs/pub27898/).
```
Comment (pwnall) [inline doc/impl.md:4]: Verified that `http://research.google.com/archive/bigtable.html` redirects to `https://research.google/pubs/pub27898/`.

**Review by pwnall — APPROVED** [review]
LGTM. Thank you for your contributions!

We'll submit a change based on this PR to the internal repository. Your PR will be merged when the change is published.

**Comment by q464pp** [comment]
可以拉取合併嗎？謝謝

**Pull request merged**